test/generaloption.py

```diff
@@ -765,6 +765,43 @@ def raise_error(msg, *args):
                               go_args=['--level-level'], envvar_prefix='GENERALOPTIONTEST', error_env_options=True,
                               error_env_option_method=raise_error)
 
+    def test_option_as_value(self):
+        """Test how valid options being used as values are handled."""
+        # -s requires an argument
+        self.mock_stderr(True)
+        self.assertErrorRegex(SystemExit, '.*', TestOption1, go_args=['-b', '-s'])
+        stderr = self.get_stderr()
+        self.assertTrue("-s option requires an argument" in stderr)
+
+        # valid ways of specifying '-b' as a value to --store/-s
+        topt = TestOption1(go_args=['--store=-b'])
+        self.assertEqual(topt.options.store, '-b')
+        self.assertFalse(topt.options.base)  # remain False (default value)
+
+        topt = TestOption1(go_args=['-s-b'])  # equivalent to --store=-b
+        self.assertEqual(topt.options.store, '-b')
+        self.assertFalse(topt.options.base)  # remain False (default value)
+
+        # -s -b is not a valid list of flags, since it's ambiguous: is '-b' a value for '-s', or an option?
+        self.mock_stderr(True)  # reset
+        self.assertErrorRegex(SystemExit, '.*', TestOption1, go_args=['-s', '-b'])
+        stderr = self.get_stderr()
+        self.assertTrue("Value '-b' is also a valid option" in stderr)
+
+        # same for --store -b
+        self.mock_stderr(True)  # reset
+        self.assertErrorRegex(SystemExit, '.*', TestOption1, go_args=['--store', '-b'])
+        stderr = self.get_stderr()
+        self.assertTrue("Value '-b' is also a valid option" in stderr)
+
+        # same for --store --base
+        self.mock_stderr(True)  # reset
+        self.assertErrorRegex(SystemExit, '.*', TestOption1, go_args=['--store', '--base'])
+        stderr = self.get_stderr()
+        self.assertTrue("Value '--base' is also a valid option" in stderr)
+
+        # FIXME: this should fail?
+        topt = TestOption1(go_args=['--store', '--nosuchoptiondefinedfoobar'])
```

Comment:
not right away, but let it fail somehow if you think it should. i'll have a look at it later.